crates/ruma-common/src/events/content.rs

```diff
@@ -104,32 +100,38 @@ pub enum EventKind {
 
 /// Content of a global account-data event.
 pub trait GlobalAccountDataEventContent:
-    EventContent<EventType = GlobalAccountDataEventType>
+    EventContentFromType<EventType = GlobalAccountDataEventType>
```

Comment:
I don't think we should have the `FromType` trait as a supertrait of these sub-traits as they are also used for the sending case, where we still use the content enums. We could consider removing the content enums altogether though...